packages/sources/gramchain/src/transport/getgrambalances.ts

```diff
@@ -0,0 +1,61 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes } from '../endpoint/getgrambalances'
+
+export type ResponseSchema = {
+  VaultID: string
+  MetalCode: string
+  CustodianID: string
+  UtilizationLockCode: string
+  EntityID: string
+  ItemCategoryCode: string
+  NrParcels: number
+  PureGrams: number
+  GrossGrams: number
+  FixedValuation: number
+  AsOfUTC: string
+  MetalName: string
+  CategoryName: string
+  ParcelGrouping: string
+  Valuation: number
+}[]
+
+export type HttpTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+export const httpTransport = new HttpTransport<HttpTransportTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        params: [param],
+        request: {
+          baseURL: config.API_ENDPOINT,
+          url: '/getgrambalances',
+          params: {
+            custodianID: param.custodianID,
+            metalCode: param.metalCode,
+            utilizationLockCode: param.utilizationLockCode,
+          },
+        },
+      }
+    })
+  },
+  parseResponse: (params, response) => {
+    return params.map((param) => {
+      const result = parseFloat(
+        response.data.reduce((sum, item) => sum + item.PureGrams, 0).toFixed(4),
+      )
```

Comment:
Seems like v2 didn't have it but just for safety could you add some validation on `item.PureGrams` to make sure it exists and if it's a number before we use it here?